File: services/frontend/react_app/src/types/rideHistory.ts
export type RideHistoryEntry = {
    sessionID: number,
    requestID: number,
    pickupLocation: string,
    dropoffLocation: string,
    requestedAt: string,
    driverPlate: string,
}

export type RideHistoryResponse = {
    totalCount: number,
    entries: RideHistoryEntry[],
}

File: services/location/ride_history.go
package location

import "time"

// RideHistoryEntry represents one requested ride persisted in storage.
type RideHistoryEntry struct {
	SessionID       uint      `json:"sessionID"`
	RequestID       uint      `json:"requestID"`
	PickupLocation  string    `json:"pickupLocation"`
	DropoffLocation string    `json:"dropoffLocation"`
	RequestedAt     time.Time `json:"requestedAt"`
	DriverPlate     string    `json:"driverPlate"`
}
